scheduler/src/cook/mesos/api.clj

```diff
@@ -1378,15 +1354,22 @@
 
 (defn check-jobs-exist
   [conn ctx]
-  (let [jobs (jobs-from-request ctx)
-        jobs-not-existing (remove (partial job-exists? (d/db conn)) jobs)]
+  (let [db (d/db conn)
+        jobs (jobs-from-request ctx)
+        jobs-not-existing (remove (partial job-exists? db) jobs)]
     (cond
       ; error case
       (seq jobs-not-existing)
       [false {::error (->> jobs-not-existing
                            (map #(str "UUID " % " does not correspond to a job."))
                            (str/join \space))}]
 
+      ; remove non-failed jobs if necessary
+      (::failed-only? ctx)
+      [true {::jobs (with-meta
```

Comment:
I'm guessing the `with-meta` thing is to deal with liberator bs?

Reply:
Yup! I'll add a comment. This prevents Liberator from *concatenating* the vectors (which is the default behavior) because we only want to keep the failed jobs in this case.

Reply:
Done. (The comment is just above the line with `:replace`.)